crates/ruma-client-api/src/error.rs

```diff
@@ -504,6 +524,76 @@ impl TryFrom<&AuthenticateError> for http::HeaderValue {
     }
 }
 
+/// How long a client should wait before they can try again.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[allow(clippy::exhaustive_enums)]
+pub enum RetryAfter {
+    /// The client should wait for the given duration.
+    ///
+    /// This variant should be preferred for backwards compatibility, as it will also populate the
+    /// `retry_after_ms` field in the body of the response.
+    Delay(Duration),
+    /// The client should wait for the date and time corresponding to the given timestamp.
+    DateTime(SecondsSinceUnixEpoch),
```

Comment:
It seems a bit weird to me to be using a `UInt`-based type here when the unix timestamp here doesn't come from JSON. Since we're now depending on `web-time` anyways, I think we can use `UNIX_EPOCH + Duration::from_secs()` and store the resulting `SystemTime` here?

Reply:
Changed to `SystemTime`.